cfme/tests/containers/test_labels.py

```diff
@@ -0,0 +1,230 @@
+    # service endpoint
+    service_endpoint = 'https://' + provider.hostname + ':8443/api/v1/namespaces/' + \
+                       service_name_in_project + '/services/' + service_name
+
+    rc_name = str(random.choice(provider.mgmt.list_replication_controller())[0])
+    rc_name_in_project = [x.project_name for x in provider.mgmt.list_replication_controller()
+                          if x[0] == rc_name][0]
+    # rc endpoint
+    rc_endpoint = 'https://' + provider.hostname + ':8443/api/v1/namespaces/' + rc_name_in_project \
+                  + '/replicationcontrollers/' + rc_name
+    # PATCH cannot be executed on a pod that is not running,
+    pods_list = provider.mgmt.list_container_group()
+    pods_all_names = [x[0] for x in pods_list]
+    pods_names_revised = filter(lambda ch: 'metrics-deployer' not in ch, pods_all_names)
+    pod_name = str(random.choice(pods_names_revised))
+    pod_name_in_project = [x.project_name for x in provider.mgmt.list_container_group()
+                           if x[0] == pod_name][0]
+    # pod endpoint
+    pod_endpoint = 'https://' + provider.hostname + ':8443/api/v1/namespaces/' + \
+                   pod_name_in_project + '/pods/' + pod_name
+
+    # picking the master node
+    node_name = provider.hostname
+    # node endpoint
+    node_endpoint = 'https://' + provider.hostname + ':8443/api/v1/nodes/' + provider.hostname
+    # template name will be picked randomly
+    template_name = str(random.choice(provider.mgmt.list_template())[0])
+    template_name_in_project = [
+        x.project_name for x in provider.mgmt.list_template() if x[0] == template_name][0]
+    # template endpoint
+    template_endpoint = 'https://' + provider.hostname + ':8443/oapi/v1/namespaces/' + \
+                        template_name_in_project + '/templates/' + template_name
+    # image name will be picked randomly
+    image_name = str(random.choice(provider.mgmt.list_image_openshift())['name'])
+    # image endpoint
+    image_endpoint = 'https://' + provider.hostname + ':8443/oapi/v1/images/' + image_name
+
+    object_names.extend([project_name, service_name, rc_name, pod_name, node_name,
+                         template_name, image_name])
+
+    obj_endpoints.extend([project_endpoint, service_endpoint, rc_endpoint, pod_endpoint,
+                          node_endpoint, template_endpoint, image_endpoint])
+
+    # use different label for each object and generate json
+    for obj_name in objects_ocp:
+        key_label = fauxfactory.gen_alphanumeric(6)
+        value_label = fauxfactory.gen_alphanumeric(6)
+        json_payload_format = '{"apiVersion":"v1","kind":' + obj_name + \
+                              ',"metadata":{"labels":{' + '"{}"'.format(key_label) + ':' \
+                              + '"{}"'.format(value_label) + '}}}'
+        json_new_payload.append(json_payload_format)
+
+    return json_new_payload, object_names, obj_endpoints
```

Comment:
@gshefer I can try, the problem is, I need to put the object names and the respective endpoints in two separate lists. I'll see if this can be implemented with a new function.